framework/contrib/hit/parse.cc

```diff
@@ -418,11 +429,43 @@ Field::path()
 }
 
 std::string
-Field::render(int indent)
+Field::render(int indent, const std::string & indent_text, int maxlen)
 {
-  std::string s = "\n" + strRepeat(indentString, indent) + _field + " = " + _val;
+  std::string s = "\n" + strRepeat(indent_text, indent) + _field + " = ";
+  size_t prefix_len = s.size() - 1;
+  auto quote = quoteChar(_val);
+  int max = maxlen - prefix_len - 1;
+  if (_kind == Kind::String && quote != "" && max > 0)
+  {
+    std::string unquoted = _val.substr(1, _val.size() - 2);
+
+    size_t pos = 0;
+    while (pos + max < unquoted.size())
+    {
+      size_t boundary = pos + max;
+      while (boundary > pos && !charIn(unquoted[boundary], " \t"))
+        boundary--;
+      if (boundary == pos)
+        boundary = pos + max;
+      boundary = std::min(boundary + 1, unquoted.size());
+
+      if (pos > 0)
+        s += "\n" + strRepeat(" ", prefix_len);
+      s += quote + unquoted.substr(pos, boundary - pos) + quote;
+      pos = boundary;
+    }
+    if (pos < unquoted.size())
+    {
+      if (pos > 0)
+        s += "\n" + strRepeat(" ", prefix_len);
+      s += quote + unquoted.substr(pos, std::string::npos) + quote;
+    }
+  }
+  else
+    s += _val;
```

Comment:
This routine is not exactly clear. How about a few comments about what's going on here.

Reply:
Done.